classes/class-wc-connect-shipping-label.php

```diff
@@ -0,0 +1,316 @@
+								),
+							),
+						),
+					),
+				),
+				'summary' => '{rate}',
+			);
+
+			$layout[] = array(
+				'type' => 'summary',
+				'tab_title' => 'Preview',
+				'title' => 'dfhdfghdfgh',
+				'action_label' => 'Print',
+				'confirmation_flag' => 'confirm',
+			);
+
+			return $layout;
+		}
+
+		protected function get_form_schema() {
+			$properties = array();
+
+			$address_fields = array(
+				'first_name' => array(
+					'type' => 'string',
+					'title' => 'First name',
+					'minLength' => 1,
+				),
+				'last_name' => array(
+					'type' => 'string',
+					'title' => 'Last name',
+				),
+				'company' => array(
+					'type' => 'string',
+					'title' => 'Company',
+				),
+				'address_1' => array(
+					'type' => 'string',
+					'title' => 'Address',
+					'minLength' => 1,
+				),
+				'address_2' => array(
+					'type' => 'string',
+					'title' => 'Address (cont.)',
+				),
+				'city' => array(
+					'type' => 'string',
+					'title' => 'City',
+					'minLength' => 1,
+				),
+				'state' => array(
+					'type' => 'string',
+					'title' => 'State',
+					'minLength' => 1,
+				),
+				'postcode' => array(
+					'type' => 'string',
+					'title' => 'Postal code',
+					'minLength' => 1,
+				),
+				'country' => array(
+					'type' => 'string',
+					'title' => 'Country',
+					'minLength' => 1,
+				),
+			);
+			$required_address_fields = array('first_name', 'address_1', 'city', 'state', 'postcode', 'country');
+
+			foreach( $address_fields as $key => $value ) {
+				$properties[ 'orig_' . $key ] = $value;
+				$properties[ 'dest_' . $key ] = $value;
+			}
+
+			$required_fields = array();
+
+			foreach( $required_address_fields as $field_name ) {
+				$required_fields[] = 'orig_' . $field_name;
+				$required_fields[] = 'dest_' . $field_name;
+			}
+
+			$itemDefinition = array(
+				'type' => 'object',
+				'required' => array(),
+				'properties' => array(
+					'height' => array(
+						'type' => 'number',
+						'title' => 'Height',
+					),
+					'product_id' => array(
+						'type' => 'string',
+						'title' => 'Product ID',
+					),
+					'length' => array(
+						'type' => 'number',
+						'title' => 'Length',
+					),
+					'quantity' => array(
+						'type' => 'number',
+						'title' => 'Quantity',
+					),
+					'weight' => array(
+						'type' => 'number',
+						'title' => 'Weight',
+					),
+					'width' => array(
+						'type' => 'number',
+						'title' => 'Width',
+					),
+				),
+			);
+
+			$properties[ 'cart' ] = array(
+				'type' => 'array',
+				'title' => 'Items to send',
+				'items' => $itemDefinition,
+			);
+			$required_fields[] = 'cart';
+
+			$properties[ 'rate' ] = array(
+				'type' => 'string',
+				'title' => 'Shipping rate',
+				'enum' => array( 'pri_1', 'media' ),
+			);
+			$required_fields[] = 'rate';
+
+			return array(
+				'required' => $required_fields,
+				'properties' => $properties,
+			);
+		}
+
+		public function meta_box( $post ) {
```

Comment:
An order may have multiple packages, and may not have been box packed yet at all (e.g. they merchant uses free shipping). Please add a TODO and an issue to wit. We are going to have to handle 1 (typical) to N (possible) labels for each order.